xarray/core/rolling.py

```diff
@@ -102,12 +128,16 @@ def __repr__(self):
             klass=self.__class__.__name__, attrs=",".join(attrs)
         )
 
-    def __len__(self):
-        return self.obj.sizes[self.dim]
+    def __len__(self) -> int:
+        return int(np.prod([self.obj.sizes[d] for d in self.dim]))
```

Comment:
again, maybe use `math.prod`?

Reply:
While at it: The previous implementation simply did not work, seems like nobody ever caught this.
Guess it was missed out when supporting multiple dims.

What is len(some rolling) even supposed to return?

Reply:
Number of windows I guess. You can iterate over a rolling object.

Reply:
Ok, iteration is only supported for 1D rolling for now.
Anyway, a possible stride argument would make it impossible to know the length.
But I guess the product of the shape is still the most correct answer.